crnk-core/src/main/java/io/crnk/core/engine/information/bean/BeanAttributeInformation.java

```diff
@@ -160,10 +148,56 @@ public String getJsonName() {
 		return jsonName.isEmpty() ? name : jsonName;
 	}
 
-	public Type getImplementationType() {
+	public Type getGenericType() {
 		if (field != null) {
 			return field.getGenericType();
 		}
-		return getter.getGenericReturnType();
+		if (getter != null) {
+			return getter.getGenericReturnType();
+		}
+		if (setter != null) {
+			final Type[] parameterTypes = setter.getGenericParameterTypes();
+			if (parameterTypes.length == 1) {
+				return parameterTypes[0];
+			}
+		}
+		throw new IllegalStateException("Required: field, getter, or setter with exactly one argument!");
 	}
+
+	/**
+	 * The {@link Member} that represents this bean attribute.
+	 */
+	protected Member getMember() {
+		if (field != null) {
+			return field;
+		}
+		if (getter != null) {
+			return getter;
+		}
+		if (setter != null) {
+			return setter;
+		}
+		throw new IllegalStateException("Required: field, getter, or setter!");
+	}
+
+	/**
+	 * The {@link AccessibleObject} that represents this bean attribute.
+	 */
+	protected AccessibleObject getAccessibleObject() {
+		if (field != null) {
+			return field;
+		}
+		if (getter != null) {
+			return getter;
+		}
+		if (setter != null) {
+			return setter;
+		}
+		throw new IllegalStateException("Required: field, getter, or setter!");
+	}
+
+	public Class<?> resolveRawClass() {
+		return TypeResolver.resolveRawClass(this.getGenericType(), this.getBeanInformation().getImplementationClass());
```

Comment:
could you cache this? BeanInformation must be efficient to look-up for performance reasons.